src/fun_lti_provider/lti.py

```diff
@@ -0,0 +1,196 @@
+
+        if self._valid is not True:
+            raise LTIException("LTI verification failed")
+
+        self._params = launch_params
+        self._verified = True
+
+        return self._valid
+
+    @property
+    def is_valid(self) -> bool:
+        """Check if the LTI request is verified and valid
+
+        Returns:
+            True if the request is verified and valid
+        """
+        return self._verified and self._valid
+
+    def get_param(self, name: str, default: Any = None):
+        """Retrieve a LTI parameter value given its name
+
+        Args:
+            name: Name of the LTI parameter
+            default: Default value if the parameter is not found
+
+        Returns:
+            The value of the LTI parameter if it exists, or the default value otherwise.
+
+        """
+        if not self._verified:
+            raise LTIRequestNotVerifiedException()
+        return self._params.get(name, default)
+
+    def get_consumer(self) -> LTIConsumer:
+        """Retrieve the LTI consumer that initiated the launch request"""
+        consumer_key = self.get_param("oauth_consumer_key")
+        passport = LTIPassport.objects.get(
+            oauth_consumer_key=consumer_key, is_enabled=True
+        )
+        return passport.consumer
+
+    def get_course_info(self) -> dict:
+        """Retrieve course info in the LTI request.
+
+        Returns
+        -------
+        dict
+            school_name: the school name
+            course_name: the course name
+            course_section: the course section
+
+        """
+        if self.is_edx_format:
+            groups = re.match(r"^course-v[0-9]:(.*)", self.get_param("context_id"))
+            if groups is not None:
+                part = groups.group(1).split("+")
+                length = len(part)
+                return {
+                    "school_name": part[0] if length >= 1 else None,
+                    "course_name": part[1] if length >= 2 else None,
+                    "course_run": part[2] if length >= 3 else None,
+                }
+
+        return {
+            "school_name": self.get_param("tool_consumer_instance_name", None),
+            "course_name": self.get_param("context_title"),
+            "course_run": None,
+        }
+
+    @property
+    def resource_link_title(self) -> Optional[str]:
+        """Return the resource link id as default for its title."""
+        return self.get_param("resource_link_title", self.get_param("resource_link_id"))
+
+    @property
+    def context_title(self) -> Optional[str]:
+        """Return the context id as default for its title."""
+        return self.get_param("context_title", self.get_param("context_id"))
+
+    @property
+    def roles(self):
+        """LTI roles of the authenticated user.
+
+        Returns
+        -------
+        set
+            normalized LTI roles from the session
+
+        """
+        roles = self.get_param("roles", [])
+        return [*map(str.lower, roles)]
```

Comment:
why not?
```suggestion
        return list(map(str.lower, roles))
```

Reply:
I hesitated with your suggestion, but I found the syntax with the unpacking operator easier to read.

But I can change :slightly_smiling_face:

Reply:
Oh then! I think `list` is more pythonic.
I recently read that Python 3.9 introduces a | operator to merge dictionaries and replace the ** notation...
But I realized that js developpers love the star notation and it's fined! :man_shrugging: